File: C++/Integral Points Inside Triangle.cpp
class Solution {
  public:
    long long int gcd (long long int a,long long int b){
        if(a<b){
            return gcd(b,a);
        }
        if(b){
            return gcd(b, a%b);
        }
        return a;
    }
    long long int InternalCount(long long int p[], long long int q[],
                                long long int r[]) {
        long long int Area = abs(p[0]*(q[1] - r[1]) + q[0]*(r[1] - p[1]) + r[0] * (p[1] - q[1]))/2;
        long long int boundaryPoints = gcd(abs(p[0]-q[0]), abs(p[1]-q[1])) + gcd(abs(q[0]-r[0]),abs(q[1]-r[1])) + gcd(abs(p[0]-r[0]), abs(p[1] - r[1]));
        long long int InternalPoints = Area + 1 - (boundaryPoints/2) ;
        return InternalPoints;
        
    }
};

File: JAVA/Integral Points Inside Triangle.java
class Solution {
    long InternalCount(long[] point1, long[] point2, long[] point3) {
        long area = Math.abs(point1[0] * (point2[1] - point3[1]) + 
                             point2[0] * (point3[1] - point1[1]) + 
                             point3[0] * (point1[1] - point2[1]));
        long boundarySum = calculateGCD(point1, point2) + 
                           calculateGCD(point2, point3) + 
                           calculateGCD(point3, point1);
        return (area - boundarySum + 2) / 2;
    }
    long calculateGCD(long[] a, long[] b) {
        return computeGCD(Math.abs(a[0] - b[0]), Math.abs(a[1] - b[1]));
    }
    long computeGCD(long p1, long p2) {
        while (p2 != 0) {
            long temp = p2;
            p2 = p1 % p2;
            p1 = temp;
        }
        return p1;
    }
}
